### PR TITLE
Add manifold for negative binomial 

### PR DESCRIPTION
**Pull request by ismailsenoz**
This PR adds manifold for negative binomial parameters.

### CODE REVIEW

File: src/natural_manifolds/negative_binomial.jl [hunk 1]
```diff
@@ -0,0 +1,19 @@
+"""
+    get_natural_manifold_base(::Type{NegativeBinomial}, ::Tuple{}, conditioner=nothing)
+
+Get the natural manifold base for the `NegativeBinomial` distribution.
+"""
+function get_natural_manifold_base(::Type{NegativeBinomial}, ::Tuple{}, conditioner=nothing)
+    @assert conditioner >= 0 "Conditioner should be negative"
```
[the remark this answers is not in the document]
Comment (ismailsenoz) [reply src/natural_manifolds/negative_binomial.jl:7]: :D positive

**Review by bvdmitri — APPROVED** [review]

**Comment by bvdmitri** [comment]
The check in `ExponentialFamily.jl` must be adjusted then

**Comment by ismailsenoz** [comment]
I think the check in ExponentialFamily is correct. It was my bad.

**Comment by bvdmitri** [comment]
Then the check in this PR must be changed :D

**Pull request merged**